funclib/kubectl/pod_node_affinity.yml

```diff
@@ -0,0 +1,40 @@
+---
+
+# This utilty task file check for node affinity for the POD
+# This utility task uses pod_name and namespace as variables.
+
+
+- name: Getting the Node name for the given pod
+  shell: kubectl get pod {{ pod_name }} -n {{ ns }} -o jsonpath='{.spec.nodeName}'
+  register: node_name
+
+- name: Cordoning the Node
+  shell: kubectl cordon {{ node_name.stdout }}  
+
+- name: Checking whether Node is cordoned
+  shell: kubectl get nodes {{ node_name.stdout }} -o jsonpath='{.spec.unschedulable}'
+  register: result
+  failed_when: "'true' not in result.stdout" 
+
+- name: Deleting the specified pod
+  shell: kubectl delete pod {{ pod_name }} -n {{ ns }}
+
+- name: Getting the Pod name in Pending state
+  shell: kubectl get pods -n {{ ns }} -o jsonpath='{.items[?(@.status.phase=="Pending")].metadata.name}'
```

Comment:
include pod label in the condition. Just to eliminate the listing of any other pod in pending state.